lib/rucio/core/replica.py

```diff
@@ -1957,7 +2040,8 @@ def __cleanup_after_replica_deletion(scope_name_temp_table, scope_name_temp_tabl
     # Update incomplete state
     messages, dids_to_delete = [], set()
     if incomplete_dids:
-        session.query(scope_name_temp_table).delete()
+        stmt = delete(scope_name_temp_table)
+        session.execute(stmt)
         session.execute(insert(scope_name_temp_table), [sn._asdict() for sn in incomplete_dids])
```

Comment:
Re: https://github.com/rucio/rucio/pull/6591/files#r1549891959

Reply:
Changed to:

```
        session.execute(stmt)
        values = [sn._asdict() for sn in incomplete_dids]
        stmt = insert(scope_name_temp_table)
```